src/ext/UtilExtension/wixext/UtilCompiler.cs

```diff
@@ -2683,7 +2683,7 @@ private void ParseProductSearchElement(XElement node)
                 }
 
                 Row row = this.Core.CreateRow(sourceLineNumbers, "WixProductSearch");
-                row[0] = id;
+                row[0] = id.Id;
```

Comment:
This is close. The `id` should be passed into the `CreateRow()` method and `row[0]` should not be set.